Fix crash when tablename or dbname is not set

SeanMollet opened the pull request:
### Description

Drop statements don't give proper tablenames, resulting in a crash. Obviously, that issue needs fixed as well, but handling it cleanly seems preferable.

Signed-off-by: Sean Mollet <sean@malmoset.com>

Fixes #

Before submitting pull request, please review the following checklist:

- [X] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [X] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [X] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [X] Every commit has a descriptive commit message.
- [X] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [X] Any new functionality is covered by tests.

williamdes commented on libraries/classes/Tracker.php:
```diff
@@ -125,7 +125,7 @@ protected static function getTableName($string)
     public static function isTracked($dbname, $tablename)
     {
-        if (! self::$enabled) {
+        if (! self::$enabled || !isset($dbname) || !isset($tablename)) {
```
```suggestion
        if (! self::$enabled || ! empty($dbname) || ! empty($tablename)) {
```

SeanMollet committed:
Fix crash when tablename or dbname is not set

Drop statements don't give proper tablenames, resulting in a crash. Obviously, that issue needs fixed as well, but handling it cleanly seems preferable.

Signed-off-by: Sean Mollet <sean@malmoset.com>

williamdes commented:
Thank you for the change

williamdes commented:
Can you post a way to reproduce this issue?

SeanMollet commented:
Sure. Create 2 tables then try the following in a SQL window:
drop table 1;
drop table 2;

For me at least, using the following:
 nginx/1.15.3
Database client version: libmysql - mysqlnd 5.0.12-dev - 20150407 - $Id: 38fea24f2847fa7519001be390c98ae0acafe387 $
PHP extension: mysqli Documentation curl Documentation mbstring Documentation
PHP version: 7.2.9-1+0~20180901081133.4+stretch~1.gbpda
 8.0.16-7 - Percona Server (GPL), Release 7, Revision 613e312 

I get the following in the error_log:
2020/01/30 14:03:52 [error] 18#18: *1 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught TypeError: Argument 1 passed to PhpMyAdmin\DatabaseInterface::escapeString() must be of the type string, null given, called in phpadmin/libraries/classes/Tracker.php on line 149 and defined in phpadmin/libraries/classes/DatabaseInterface.php:2997
Stack trace:
#0 phpadmin/libraries/classes/Tracker.php(149): PhpMyAdmin\DatabaseInterface->escapeString(NULL)
#1 phpadmin/libraries/classes/Tracker.php(841): PhpMyAdmin\Tracker::isTracked('switcheon', NULL)
#2 phpadmin/libraries/classes/DatabaseInterface.php(345): PhpMyAdmin\Tracker::handleQuery('drop table if e...')
#3 phpadmin/libraries/classes/Sql.php(958): PhpMyAdmin\DatabaseInterface->tryQuery('drop table if e...', 256, 1)
#4 phpadmin/libraries/classes/Sql.p" while reading response header from upstream, client: 10.0.10.1, server: danebramage.org, request: "POST /phpadmin/import.php HTTP/2.0", upstream: "fastcgi://unix:/run/php/php7.2-fpm.sock:", host: "danebramage.org"

williamdes commented:
Thank you!
I will try that as soon as I am home, if someone can test and approve that would be awesome

yashrajbothra commented:
I tried to reproduce on demo but it turned out fine for me.
![image](https://user-images.githubusercontent.com/47683556/73596637-f1ce9c00-4549-11ea-81dd-1fc19c6e7112.png)

williamdes commented:
Thank you @yashrajbothra for testing that
If someone can research in the codebase what can give a null parameter that would help but for now I can research on mobile :)

williamdes commented:
@SeanMollet would this be an option to fix it ?
 ```patch
diff --git a/libraries/classes/Tracker.php b/libraries/classes/Tracker.php
index e8e5cb0525f..756232f1843 100644
--- a/libraries/classes/Tracker.php
+++ b/libraries/classes/Tracker.php
@@ -917,10 +917,10 @@ class Tracker
                 // we want to track
                 $sql_query .=
                 " WHERE FIND_IN_SET('" . $result['identifier'] . "',tracking) > 0" .
-                " AND `db_name` = '" . $GLOBALS['dbi']->escapeString($dbname) . "' " .
+                " AND `db_name` = '" . $GLOBALS['dbi']->escapeString($dbname ?? '') . "' " .
                 " AND `table_name` = '"
                 . $GLOBALS['dbi']->escapeString($result['tablename']) . "' " .
-                " AND `version` = '" . $GLOBALS['dbi']->escapeString($version) . "' ";
+                " AND `version` = '" . $GLOBALS['dbi']->escapeString($version ?? '') . "' ";
 
                 $relation->queryAsControlUser($sql_query);
             }
```

williamdes commented:
Thank you for your contribution @SeanMollet !
